assets/src/design-system/components/slider/index.js

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import { useRef, useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useKeyDownEffect } from '../keyboard';
+import { Tooltip } from '../tooltip';
+import { PLACEMENT } from '../popup';
+
+const DEFAULT_SIZE = 24;
+const rangeThumb = css`
+  appearance: none;
+  width: ${({ thumbSize }) => thumbSize}px;
+  height: ${({ thumbSize }) => thumbSize}px;
+  background-color: ${({ theme }) => theme.colors.interactiveBg.primaryNormal};
+  cursor: pointer;
+  border-radius: 50px;
+`;
+const rangeThumbHover = css`
+  background-color: ${({ theme }) => theme.colors.interactiveBg.primaryHover};
+`;
+
+const Input = styled.input.attrs({
+  type: 'range',
+})`
+  position: relative;
+  min-width: 100px;
+  cursor: pointer;
+  outline: none;
+  background: ${({ theme }) => theme.colors.bg.quaternary};
+  border-radius: 100px;
+  height: 4px;
+  appearance: none;
+  flex: 1;
+  margin: 0;
+
+  ::-webkit-slider-thumb {
+    ${rangeThumb}
+  }
+
+  ::-moz-range-thumb {
+    ${rangeThumb}
+  }
+
+  ::-ms-thumb {
+    ${rangeThumb}
+  }
+
+  ::before {
+    position: absolute;
+    content: ' ';
+    height: 6px;
+    top: calc(50% - 3px);
+    left: -calc(
+      ${({ percentage, thumbSize, width }) =>
+          getAdjustedWidthPercentage(percentage, thumbSize, width)}% - 4px
+    );
+    width: ${({ percentage = 0 }) => percentage}%;
+    background-color: ${({ theme }) =>
+      theme.colors.interactiveBg.primaryNormal};
+    border-radius: 50px;
+  }
+
+  :hover {
+    ::before {
+      background-color: ${({ theme }) =>
+        theme.colors.interactiveBg.primaryHover};
+    }
+    ::-webkit-slider-thumb {
+      ${rangeThumbHover}
+    }
+
+    ::-moz-range-thumb {
+      ${rangeThumbHover}
+    }
+
+    ::-ms-thumb {
+      ${rangeThumbHover}
+    }
+  }
+
+  :focus::after {
+    position: absolute;
+    box-sizing: border-box;
```

Comment:
Only necessary until we do a root `*, *::before, *::after { box-sizing: border-box; }` on Storybook as we have in both editor and dashboard.